codec-redis/src/main/java/io/netty/handler/codec/redis/RefCountedBulkStringRedisMessage.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.DefaultByteBufHolder;
+import io.netty.util.internal.StringUtil;
+
+/**
+ * Bulk Strings of <a href="http://redis.io/topics/protocol">RESP</a>
+ */
+public class RefCountedBulkStringRedisMessage extends DefaultByteBufHolder implements BulkStringRedisMessage {
+
+    RefCountedBulkStringRedisMessage(ByteBuf content) {
+        super(content.retain());
```

Comment:
I'm not sure I would do the retain here.. I think I would prefer to have the caller have the responsibility to retain it first. This is because the caller will also need to ensure it is duplicated or sliced anyway most of the time